it/it-tests/src/test/java/it/settings/PropertySetsTest.java

```diff
@@ -22,76 +22,125 @@
-    new SeleneseTest(
-      Selenese.builder().setHtmlTestsInClasspath("create-auto-generated",
-        "/settings/PropertySetsTest/auto-generated/create.html"
-        ).build()).runOn(orchestrator);
+  public void delete_property_set() throws Exception {
+    setProperty("sonar.test.jira.servers", "jira1");
+    setProperty("sonar.test.jira.servers.jira1.url", "http://jira1");
+    setProperty("sonar.test.jira.servers.jira1.port", "12345");
 
-    String[] keys = getProperty("sonar.autogenerated").split("[,]");
-    assertThat(getProperty("sonar.autogenerated." + keys[0] + ".value")).isEqualTo("FIRST");
-    assertThat(getProperty("sonar.autogenerated." + keys[1] + ".value")).isEqualTo("SECOND");
-    assertThat(getProperty("sonar.autogenerated." + keys[2] + ".value")).isEqualTo("THIRD");
+    resetSetting("sonar.test.jira.servers");
 
-    new SeleneseTest(Selenese.builder().setHtmlTestsInClasspath("update-auto-generated",
-      "/settings/PropertySetsTest/auto-generated/update.html"
-      ).build()).runOn(orchestrator);
+    assertThat(settingsService().values(ValuesRequest.builder().setKeys("sonar.test.jira.servers").build()).getSettingsList()).isEmpty();
+  }
+
+  private void assertPropertySet(String baseSettingKey, List<Map.Entry<String, String>>... fieldsValues) {
+    Settings.Setting setting = getSetting(baseSettingKey);
+    assertThat(setting.getFieldsValues().getFieldsValuesList()).hasSize(fieldsValues.length);
+    int index = 0;
+    for (Settings.FieldsValues.Value fieldsValue : setting.getFieldsValues().getFieldsValuesList()) {
+      assertThat(fieldsValue.getValue()).containsOnly(fieldsValues[index].toArray(new Map.Entry[]{}));
+      index++;
+    }
+  }
 
-    keys = getProperty("sonar.autogenerated").split("[,]");
-    assertThat(getProperty("sonar.autogenerated." + keys[0] + ".value")).isEqualTo("FIRST");
-    assertThat(getProperty("sonar.autogenerated." + keys[1] + ".value")).isEqualTo("THIRD");
+  private Settings.Setting getSetting(String key) {
+    Settings.ValuesWsResponse response = settingsService().values(ValuesRequest.builder().setKeys(key).build());
+    List<Settings.Setting> settings = response.getSettingsList();
+    assertThat(settings).hasSize(1);
+    return settings.get(0);
   }
 
+  /**
+   * @deprecated Replace with api/settings/set WS when setting property set will be possible in the WS
+   */
+  @Deprecated
   static void setProperty(String key, String value) {
     orchestrator.getServer().getAdminWsClient().update(new PropertyUpdateQuery(key, value));
   }
 
-  static String getProperty(String key) {
-    return orchestrator.getServer().getAdminWsClient().find(new PropertyQuery().setKey(key)).getValue();
+  static void resetSetting(String... keys){
+    for (String key : keys){
+      settingsService().reset(ResetRequest.builder().setKey(key).build());
+    }
+  }
+
+  private static SettingsService settingsService() {
+    return newAdminWsClient(orchestrator).settingsService();
```

Comment:
each time you call `settingsService` you create a new admin ws client. I suggest to have the admin ws client as a parameter